ruby/lib/item_type.rb

```diff
@@ -0,0 +1,35 @@
+def aged_brie(item)
+  if quality_limit?(item)
+    increase_quality(item)
+  end
+  decrease_sell_in(item)
+  if passed_sell_in?(item) && quality_limit?(item)
+    increase_quality(item)
+  end 
+end
+  
+def backstage_passes(item)
+  if quality_limit?(item)
+    increase_quality(item)
+    if between_ten_and_six_days?(item) && quality_limit?(item)
+      increase_quality(item)
+    end
+    if between_five_and_one_days?(item) && quality_limit?(item)
+      increase_quality(item)
+    end
+  end
```

Comment:
Let's discuss live and play around live.  I'm curious if all this nesting is really necessary.  Maybe.  I dunno.